### PR TITLE
ZCS-643: Custom checkbox style

### PR DESCRIPTION
**Pull request by kalpeshhpatel**
Changes:
* DwtCheckbox.js: Added custom checkbox styling changes

This changes are covered as part of task https://jira.corp.synacor.com/browse/ZCS-643.

https://jira.corp.synacor.com/browse/ZCS-643

### CODE REVIEW

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js [hunk 1]
```diff
@@ -349,3 +378,13 @@ function(evt) {
...
 };
+
+
+/**
+ * Set custom icon for checkbox
+ */
+DwtCheckbox.prototype.__setCustomCheckboxIcon = function () {
+	var icon = this.isSelected() ? 'CheckBoxChecked' : 'CheckBox';
+	this._inputIconEl.innerHTML = AjxImg.getImageHtml(DwtCheckbox.CUSTOM_STYLE_ICONS[icon]);
+	this._inputIconEl.setAttribute(AjxEnv.isIE ? "htmlFor" : "for", this._inputEl.id);
```
Comment (silentsakky) [inline WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: do we need to set this attribute everytime when checkbox state  changes?

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js [hunk 2]
```diff
@@ -349,3 +378,13 @@ function(evt) {
 	checkbox.focus();
 	checkbox.notifyListeners(DwtEvent.SELECTION, selEv);
 };
+
+
+/**
+ * Set custom icon for checkbox
+ */
+DwtCheckbox.prototype.__setCustomCheckboxIcon = function () {
```
Comment (tarangkhandelwal) [inline WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: "function" should start from next line.

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js [hunk 3]
```diff
@@ -349,3 +378,13 @@ function(evt) {
 	checkbox.focus();
 	checkbox.notifyListeners(DwtEvent.SELECTION, selEv);
 };
+
```
Comment (tarangkhandelwal) [inline WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js:382]: extra space here? Ignore if intended.

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js [hunk 4]
```diff
@@ -317,13 +336,23 @@ function(templateId, data) {
     data.value = this._initValue;
 	data.type = this.INPUT_TYPE;
 	DwtControl.prototype._createHtmlFromTemplate.call(this, templateId, data);
+	//set custom style for check box only
+	this.CUSTOM_CHECKBOX_STYLE = DwtCheckbox.CUSTOM_CHECKBOX && this.INPUT_TYPE == 'checkbox';
```
Comment (tarangkhandelwal) [inline WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: Just to confirm: DwtCheckbox.CUSTOM_CHECKBOX  will always be true as we set it true as default. Should this be this.CUSTOM_CHECKBOX for any child widget inheriting this to disabled the custom behavior?

**Review by tarangkhandelwal — CHANGES_REQUESTED** [review]

**Review by tarangkhandelwal — CHANGES_REQUESTED** [review]
Looks good to me . Just small changes and this is good to go.

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js (hunk 4 above)
Comment (kalpeshhpatel) [reply WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: Good point. Will update it. Thanks.

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js (hunk 3 above)
Comment (kalpeshhpatel) [reply WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js:382]: Fixed: https://github.com/Zimbra/zm-ajax/pull/4/files#diff-5d22f86ec071d8b2e4f423ae9803c185R382

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js (hunk 1 above)
Comment (kalpeshhpatel) [reply WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: No, not needed, I misplaced it. 
Updated code: https://github.com/Zimbra/zm-ajax/pull/4/files#diff-5d22f86ec071d8b2e4f423ae9803c185R355 Thanks for pointing it though.

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js (hunk 4 above)
Comment (kalpeshhpatel) [reply WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: Done: https://github.com/Zimbra/zm-ajax/pull/4/files#diff-5d22f86ec071d8b2e4f423ae9803c185R111 and https://github.com/Zimbra/zm-ajax/pull/4/files#diff-5d22f86ec071d8b2e4f423ae9803c185R340

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js (hunk 2 above)
Comment (kalpeshhpatel) [reply WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js]: Done: https://github.com/Zimbra/zm-ajax/pull/4/files#diff-5d22f86ec071d8b2e4f423ae9803c185R387

File: WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js [hunk 5]
```diff
@@ -317,13 +336,24 @@ function(templateId, data) {
     data.value = this._initValue;
 	data.type = this.INPUT_TYPE;
 	DwtControl.prototype._createHtmlFromTemplate.call(this, templateId, data);
+	//set custom style for check box only
+	this.CUSTOM_CHECKBOX_STYLE = this.CUSTOM_CHECKBOX && this.INPUT_TYPE == 'checkbox';
```
Comment (SatyamKumarOnline) [inline WebRoot/js/ajax/dwt/widgets/DwtCheckbox.js:340]: Can we have === , here

**Review by tarangkhandelwal — APPROVED** [review]
Looks good to me.

**Review by SatyamKumarOnline — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Pull request merged**